opm/output/eclipse/EclipseWriter.cpp

```diff
@@ -580,880 +354,418 @@ class Init : private boost::noncopyable
 
 
         // finally, write the grid to disk
-        IOConfigConstPtr ioConfig = eclipseState->getIOConfigConst();
-        if (ioConfig->getWriteEGRIDFile()) {
-            if (eclipseState->getDeckUnitSystem().getType() == UnitSystem::UNIT_TYPE_METRIC){
-                eclGrid->fwriteEGRID(egridFileName_.ertHandle(), true);
-            }else{
-                eclGrid->fwriteEGRID(egridFileName_.ertHandle(), false);
-            }
+        const auto& ioConfig = *es.getIOConfig();
+        if( ioConfig.getWriteEGRIDFile() ) {
+            const bool is_metric =
+                es.getDeckUnitSystem().getType() == UnitSystem::UNIT_TYPE_METRIC;
+            eclGrid->fwriteEGRID( egrid.ertHandle(), is_metric );
         }
 
 
-        if (ioConfig->getWriteINITFile()) {
-            Keyword<float> poro_kw("PORO", dataField);
-            ecl_init_file_fwrite_header(ertHandle(),
-                                        eclGrid->c_ptr(),
-                                        poro_kw.ertHandle(),
-                                        ertPhaseMask(uses),
-                                        timer.currentPosixTime());
+        if( ioConfig.getWriteINITFile() ) {
+            Keyword< float > poro_kw( "PORO", dataField );
+            ecl_init_file_fwrite_header( this->ertHandle(),
+                                         eclGrid->c_ptr(),
+                                         poro_kw.ertHandle(),
+                                         ert_phase_mask,
+                                         current_posix_time );
         }
     }
 
-    void writeKeyword(const std::string& keywordName, const std::vector<double> &data)
-    {
-        Keyword <float> kw(keywordName, data);
-        ecl_kw_fwrite(kw.ertHandle(), ertHandle());
+    void writeKeyword( const std::string& keywordName,
+                       const std::vector<double> &data ) {
+        Keyword< float > kw( keywordName, data );
+        ecl_kw_fwrite( kw.ertHandle(), this->ertHandle() );
     }
 
-    fortio_type *ertHandle() const
-    { return ertHandle_; }
+    fortio_type* ertHandle() { return this->init.get(); }
 
 private:
-    fortio_type *ertHandle_;
-    FileName egridFileName_;
+    FileName egrid;
+    ERT::ert_unique_ptr< fortio_type, fortio_fclose > init;
 };
 
+const int inactive_index = -1;
 
+/// Convert OPM phase usage to ERT bitmask
```

Comment:
The `ECL_WATER_PHASE` and so on values are really Eclipse values and not ERT values.